dashboard/src/components/api/workspace/che-workspace.factory.ts

```diff
@@ -407,6 +407,34 @@ export class CheWorkspace {
     return defer.promise;
   }
 
+  /**
+   * Validates machine token for the workspace
+   * 
+   * @param workspaceId workspace ID
+   * @param token Che machine token
+   */
+  validateMachineToken(workspaceId: string, token: string): ng.IPromise<any> {
+    const defer = this.$q.defer();
+
+    const promise: ng.IHttpPromise<any> = this.$http.get(`/api/workspace/${workspaceId}`, {
+      params: {
+        token: token
+      }
+    });
+
+    promise.then((response: ng.IHttpPromiseCallbackArg<che.IWorkspace>) => {
+      defer.resolve();
+    }, (error: any) => {
+      if (error && error.status === 304) {
+        defer.resolve();
+        return;
```

Comment:
Should we return 'return defer.promise;' ?

Reply:
No, we only need there to resolve or reject that deferred object promise. And return it at the end of the method.